### PR TITLE
255 confirmation dialog

### LINKED ISSUE #255: [FRONTEND] Create confirmation dialog
**Is your feature request related to a problem? Please describe.**

Image

**BEFORE MERGING**

- [x] Storybooks created where possible
- [ ] Tested with [nock](https://tanstack.com/query/v3/docs/framework/react/guides/testing) if data fetching
- [ ] Tests written for critical interactions
- [ ] PR Reviewed (For non-trivial changes)
- [ ] Changes tested after rebasing on master or merging in master (_hint_: `git fetch origin master:master`, then `git rebase master` or `git merge master`)
- [ ] All required PR checks passing

### PR DESCRIPTION
**Pull request by zlrkw11**
Closes #255

### CODE REVIEW

File: client/src/components/generic/ConfirmationDialog/ConfirmationDialog.test.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+test("Calls the correct function upon clicking on the buttons", async () => {
+  const leftFakeFunction = jest.fn()
+  const rightFakeFunction = jest.fn()
+
+  render(
+    <ConfirmationDialog
+      title="w"
+      text="w"
+      left="ray"
```
Comment (bcho892) [inline client/src/components/generic/ConfirmationDialog/ConfirmationDialog.test.tsx:12]: nit: not good idea to put own name in code

File: client/src/components/generic/ConfirmationDialog/ConfirmationDialog.story.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    text: {
+      name: "text for description"
+    }
+  }
+}
+
+export default meta
+type Story = StoryObj<typeof meta>
+
+export const ConfirmationDialog: Story = {
```
Comment (bcho892) [inline client/src/components/generic/ConfirmationDialog/ConfirmationDialog.story.tsx:20]: nit: we should have a story showing how to use on clicks

File: client/src/components/generic/ConfirmationDialog/ConfirmationDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
+import Button from "../FigmaButtons/FigmaButton"
+type dialogVariants = "default"
+
+export interface IDialogProps {
+  title: string
+  text: string
+  variant?: dialogVariants
+  left: string
```
Comment (bcho892) [inline client/src/components/generic/ConfirmationDialog/ConfirmationDialog.tsx:8]: what does `left` and `right` mean? make it more clear or add jsdoc explaining it?

**Review by bcho892 — CHANGES_REQUESTED** [review]

File: client/src/components/generic/ConfirmationDialog/ConfirmationDialog.tsx (hunk 3 above)
Comment (zlrkw11) [reply client/src/components/generic/ConfirmationDialog/ConfirmationDialog.tsx:8]: changed it

**Review by bcho892 — APPROVED** [review]

File: client/src/components/generic/ConfirmationDialog/ConfirmationDialog.story.tsx (hunk 2 above)
Comment (zlrkw11) [reply client/src/components/generic/ConfirmationDialog/ConfirmationDialog.story.tsx:20]: done

**Pull request merged**